src/Scientist/Internals/Experiment.cs

```diff
@@ -18,15 +18,17 @@ internal class Experiment<T> : IExperiment<T>, IExperimentAsync<T>
         private readonly Dictionary<string, Func<Task<T>>> _candidates;
         private Func<T, T, bool> _comparison = DefaultComparison;
         private Func<Task> _beforeRun;
+        private readonly ILaboratory _laboratory;
         private Action<Operation, Exception> _thrown = _alwaysThrow;
         private Func<Task<bool>> _runIf = _alwaysRun;
         private readonly List<Func<T, T, Task<bool>>> _ignores = new List<Func<T, T, Task<bool>>>();
         private readonly Dictionary<string, dynamic> _contexts = new Dictionary<string, dynamic>();
 
-        public Experiment(string name)
+        public Experiment(string name, ILaboratory laboratory)
```

Comment:
What do you think about making this parameter a `Func<Task<bool>>` (and perhaps a `Func<bool>` overload too) instead of introducing this interface. I imagine in most cases, when I want to disable a set of tests, I won't want to create a whole implementation. I'll just create a lambda expression.